modules/joex/src/main/scala/docspell/joex/scanmailbox/ScanMailboxTask.scala

```diff
@@ -167,6 +170,19 @@ object ScanMailboxTask {
           .map(_.toRight(new Exception(s"Folder '$name' not found")))
           .mapF(_.rethrow)
 
+    def searchMailsRecursively[C](
+        a: Access[F, C]
+    )(folder: MailFolder): MailOp[F, C, SearchResult[MailHeader]] =
+      for {
+        subFolders <- a.listFoldersRecursive(Some(folder))
+        search <- subFolders.foldLeft(searchMails(a)(folder)) { (result, folder) =>
+          for {
+            res <- result
+            search <- searchMails(a)(folder)
+          } yield SearchResult(res.mails ++ search.mails, res.count + search.count)
```

Comment:
nit picking, only fyi: maybe this can be simplified using `traverse` (from cats). 
```scala
for {
  subfolders <- a.listFolders(Some(f))
  res <- subfolders.traverse(searchMails(a))
} yield res.foldMonoid
```
but requires to extend the `SearchResult` a bit (moving the combining stuff into a monoid instance)

Reply:
I added the combining stuff here: https://github.com/eikek/emil/pull/322. I hope I used the correct things here, since you wrote `res.foldMonoid` here and I only found the `.combineAll` function from `cats.Monoid`.